group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroupMember.java

```diff
@@ -0,0 +1,560 @@
+     * definition of the fields.
+     */
+    public static class Builder {
+        private final String memberId;
+        private Optional<String> groupInstanceId = Optional.empty();
+        private String clientId = "";
+        private String clientHost = "";
+        private int rebalanceTimeoutMs = -1;
+        private int sessionTimeoutMs = -1;
+        private String protocolType = "";
+        private List<Protocol> supportedProtocols = Collections.emptyList();
+        private byte[] assignment = new byte[0];
+
+        public Builder(String memberId) {
+            this.memberId = Objects.requireNonNull(memberId);
+        }
+
+        public Builder(GenericGroupMember member) {
+            Objects.requireNonNull(member);
+
+            this.memberId = member.memberId;
+            this.groupInstanceId = member.groupInstanceId;
+            this.rebalanceTimeoutMs = member.rebalanceTimeoutMs;
+            this.sessionTimeoutMs = member.sessionTimeoutMs;
+            this.clientId = member.clientId;
+            this.clientHost = member.clientHost;
+            this.protocolType = member.protocolType;
+            this.supportedProtocols = member.supportedProtocols;
+            this.assignment = member.assignment;
+        }
+
+        public Builder setGroupInstanceId(Optional<String> groupInstanceId) {
+            this.groupInstanceId = groupInstanceId;
+            return this;
+        }
+
+        public Builder setClientId(String clientId) {
+            this.clientId = clientId;
+            return this;
+        }
+
+        public Builder setClientHost(String clientHost) {
+            this.clientHost = clientHost;
+            return this;
+        }
+
+        public Builder setRebalanceTimeoutMs(int rebalanceTimeoutMs) {
+            this.rebalanceTimeoutMs = rebalanceTimeoutMs;
+            return this;
+        }
+
+        public Builder setSessionTimeoutMs(int sessionTimeoutMs) {
+            this.sessionTimeoutMs = sessionTimeoutMs;
+            return this;
+        }
+
+        public Builder setProtocolType(String protocolType) {
+            this.protocolType = protocolType;
+            return this;
+        }
+
+        public Builder setSupportedProtocols(List<Protocol> protocols) {
+            this.supportedProtocols = protocols;
+            return this;
+        }
+
+        public Builder setAssignment(byte[] assignment) {
+            this.assignment = assignment;
+            return this;
+        }
+
+        public GenericGroupMember build() {
+            return new GenericGroupMember(
+                memberId,
+                groupInstanceId,
+                clientId,
+                clientHost,
+                rebalanceTimeoutMs,
+                sessionTimeoutMs,
+                protocolType,
+                supportedProtocols,
+                assignment
+            );
+        }
+    }
+
+    private static class MemberSummary {
```

Comment:
Could we also use a auto-generate class to replace this?

Reply:
The reason that i kept the class is because we use this to encapsulate information for both
`ListGroups` and `DescribeGroup` apis. i.e. this is used in https://github.com/apache/kafka/pull/13663/files#diff-59d730ac42eb65d0a800f16cd34dc14fab8efbb8d70f7c220d1d3ecb82b412a6R663

The alternative would be to to just use the `GenericGroupMember` and `GenericGroup` objects instead. If you think this is better, I can create a jira to refactor this once we merge both to trunk.

Reply:
got it. i'll remove it in this PR